### PR TITLE
BlobServiceClient::CreateBlobContainer/DeleteBlobContainer/UndeleteBl…

### PR DESCRIPTION
**Pull request by Jinming-Hu**
…obContainer

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

closes https://github.com/Azure/azure-sdk-for-cpp/issues/1144 
closes https://github.com/Azure/azure-sdk-for-cpp/issues/1114

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,9 @@
 - Rename `Models::DeleteSnapshotsOption::Only` to `Models::DeleteSnapshotsOption::OnlySnapshots`.
 - Rename `SourceConditions` in API options to `SourceAccessConditions`.
 - Remove Blob Batch.
+- Add `CreateBlobContainer`, `DeleteBlobContainer`, `UndeleteBlobContainer` into `BlobServiceClient`.
+- Add `DeleteBlob` into `BlobContainerClient`.
```
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:70]: How is Adding APIs a breaking change? if it is, maybe explain it here

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp [hunk 2]
```diff
@@ -324,6 +310,19 @@ namespace Azure { namespace Storage { namespace Blobs {
     Azure::Core::Response<Models::BreakBlobContainerLeaseResult> BreakLease(
         const BreakBlobContainerLeaseOptions& options = BreakBlobContainerLeaseOptions()) const;
 
+    /**
+     * @brief Marks the specified blob or snapshot for deletion. The blob is later deleted
+     * during garbage collection. Note that in order to delete a blob, you must delete all of its
+     * snapshots. You can delete both at the same time using DeleteBlobOptions.DeleteSnapshots.
```
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:316]: As per recent research, I noticed you can use `#` to reference the API like
```suggestion
     * snapshots. You can delete both at the same time using #Azur::Storage::Blobs::DeleteBlobOptions::DeleteSnapshots.
```

**Review by vhvb1989 — APPROVED** [review]
Please review the changelog. 
I think some changes are for `new changes` section

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:316]: Nit: there is a typo in the suggested change: `#Azure`

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp [hunk 3]
```diff
@@ -324,6 +310,19 @@ namespace Azure { namespace Storage { namespace Blobs {
...
+     *
+     * @param blobName The name of the blob to delete.
+     * @param options Optional parameters to execute this function.
+     * @return Nothing.
+     */
+    Azure::Core::Response<std::nullptr_t> DeleteBlob(
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:322]: Why are we having an inconsistent return type here?

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp [hunk 4]
```diff
@@ -179,6 +179,43 @@ namespace Azure { namespace Storage { namespace Blobs {
...
+
+    /**
+     * @brief Marks the specified blob container for deletion. The container and any blobs
+     * contained within it are later deleted during garbage collection.
+     * 
+     * @param blobContainerName The name of the container to delete.
+     * @param options Optional parameters to execute this function.
+     * @return Nothing.
+     */
+    Azure::Core::Response<std::nullptr_t> DeleteBlobContainer(
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp:202]: Why are we having an inconsistent return type here?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp:202]: explained in this issue https://github.com/Azure/azure-sdk-for-cpp/issues/1322

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp (hunk 3 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:322]: explained in this issue https://github.com/Azure/azure-sdk-for-cpp/issues/1322

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:316]: @vhvb1989 I will refine the doc in the future (after GA). I opened an issue to track that. I will keep these things unchanged for now to make this PR single-purpose.

File: sdk/storage/azure-storage-blobs/CHANGELOG.md (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md]: fixed

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp (hunk 3 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:322]: I created a PR #1331 to implement `Response<void>`. If core team accepts this PR, I will change the return type here.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp (hunk 4 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp:202]: I created a PR #1331 to implement `Response<void>`. If core team accepts this PR, I will change the return type here.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp:202]: duplicate

**Pull request merged**